crates/common_utils/src/types.rs

```diff
@@ -1247,3 +1247,129 @@ where
         self.0.to_sql(out)
     }
 }
+
+/// Domain type for connector_transaction_id
+/// Maximum length for connector's transaction_id can be 128 characters in HS DB.
+/// In case connector's use an identifier whose length exceeds 128 characters,
+/// the hash value of such identifiers will be stored as connector_transaction_id.
+/// The actual connector's identifier will be stored in a separate column -
+/// connector_transaction_data or something with a similar name.
+#[derive(Debug, Clone, PartialEq, Eq, serde::Deserialize, serde::Serialize, AsExpression)]
+#[diesel(sql_type = sql_types::Text)]
+pub enum ConnectorTransactionId {
+    /// Actual transaction identifier
+    TxnId(String),
+    /// Hashed value of the transaction identifier
+    HashedData(String),
+}
+
+impl ConnectorTransactionId {
+    /// Implementation for retrieving the inner identifier
+    pub fn get_id(&self) -> &String {
+        match self {
+            Self::TxnId(id) | Self::HashedData(id) => id,
+        }
+    }
+
+    /// Implementation for forming ConnectorTransactionId and an optional string to be used for connector_transaction_id and connector_transaction_data
+    pub fn form_id_and_data(src: String) -> (Self, Option<String>) {
+        let txn_id = Self::from(src.clone());
+        match txn_id {
+            Self::TxnId(_) => (txn_id, None),
+            Self::HashedData(_) => (txn_id, Some(src)),
+        }
+    }
+
+    /// Implementation for retrieving
+    pub fn get_txn_id<'a>(
+        &'a self,
+        txn_data: Option<&'a String>,
```

Comment:
nit: this could have been `Option<&'a str>`

Reply:
Will update in the subsequent PRs